From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Simon Gardling <titaniumtown@gmail.com>
Date: Tue, 1 Dec 2020 16:50:36 -0500
Subject: [PATCH] Snow Golem rate of fire config

The formula used to determine the amount of ticks between shots is:
    ((sqrt(distanceToTarget) / snowGolemAttackDistance) / snowGolemSnowBallModifer) * (maxShootIntervalTicks - minShootIntervalTicks) + minShootIntervalTicks

If min-shoot-interval-ticks and max-shoot-interval-ticks are both set to
0, snow golems won't shoot any snowballs.

diff --git a/src/main/java/net/minecraft/world/entity/animal/SnowGolem.java b/src/main/java/net/minecraft/world/entity/animal/SnowGolem.java
index 37ac7997016ad4b1b3ed7a448b9cba761bd96530..777efbc0d9523a0994c2e0bbccc668138857cf8c 100644
--- a/src/main/java/net/minecraft/world/entity/animal/SnowGolem.java
+++ b/src/main/java/net/minecraft/world/entity/animal/SnowGolem.java
@@ -79,7 +79,7 @@ public class SnowGolem extends AbstractGolem implements Shearable, RangedAttackM
     @Override
     protected void registerGoals() {
         this.goalSelector.addGoal(0, new org.purpurmc.purpur.entity.ai.HasRider(this)); // Purpur
-        this.goalSelector.addGoal(1, new RangedAttackGoal(this, 1.25D, 20, 10.0F));
+        this.goalSelector.addGoal(1, new RangedAttackGoal(this, level.purpurConfig().mobs.snowGolem.attack.distance, level.purpurConfig().mobs.snowGolem.attack.minShootInterval, level.purpurConfig().mobs.snowGolem.attack.maxShootInterval, level.purpurConfig().mobs.snowGolem.attack.modifier)); // Purpur
         this.goalSelector.addGoal(2, new WaterAvoidingRandomStrollGoal(this, 1.0D, 1.0000001E-5F));
         this.goalSelector.addGoal(3, new LookAtPlayerGoal(this, Player.class, 6.0F));
         this.goalSelector.addGoal(4, new RandomLookAroundGoal(this));
diff --git a/src/main/java/org/purpurmc/purpur/configuration/PurpurWorldConfiguration.java b/src/main/java/org/purpurmc/purpur/configuration/PurpurWorldConfiguration.java
index cb1afb0ee90ca4fbfcdef7cac6fa47007d4d3248..b969f748b7223eff87dd534d2716ce60d6746715 100644
--- a/src/main/java/org/purpurmc/purpur/configuration/PurpurWorldConfiguration.java
+++ b/src/main/java/org/purpurmc/purpur/configuration/PurpurWorldConfiguration.java
@@ -1030,6 +1030,14 @@ public class PurpurWorldConfiguration extends ConfigurationPart {
             public class Attributes extends ConfigurationPart {
                 public double maxHealth = 4.0D;
             }
+
+            public Attack attack;
+            public class Attack extends ConfigurationPart {
+                public int minShootInterval = 20;
+                public int maxShootInterval = 20;
+                public float modifier = 10.0F;
+                public double distance = 1.25D;
+            }
         }
 
         public Squid squid;
